src/test/java/com/google/songgame/YoutubeParserTest.java

```diff
@@ -0,0 +1,102 @@
+package com.google.songgame.data;
+
+import org.junit.Assert;
+import org.junit.rules.ExpectedException;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.Rule;
+import org.junit.Ignore;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class YoutubeParserTest {
+
+  private YoutubeParser youtubeParser;
+
+  @Rule public ExpectedException thrown = ExpectedException.none();
+
+  @Before
+  public void init() {
+    youtubeParser = new YoutubeParser();
+  }
+
+  // Test getPlaylistIdFromUrl
+  @Test
+  public void parsePlaylistUrlWithParameters() {
+    String testPlaylistUrl =
+        "https://www.youtube.com/playlist?list=PLbpi6ZahtOH7CkYdkWCsAVMcX1hoYy3TP&app=desktop";
+    String playlistId = "PLbpi6ZahtOH7CkYdkWCsAVMcX1hoYy3TP";
+    Assert.assertEquals(playlistId, youtubeParser.getPlaylistIdFromUrl(testPlaylistUrl));
+
+    testPlaylistUrl =
+        "https://www.youtube.com/playlist?list=PLbpi6ZahtOH4OiptAsB2BrGjIiITwrdqs&app=desktop";
+    playlistId = "PLbpi6ZahtOH4OiptAsB2BrGjIiITwrdqs";
+    Assert.assertEquals(playlistId, youtubeParser.getPlaylistIdFromUrl(testPlaylistUrl));
+  }
+
+  @Test
+  public void parseYoutubeShareLink() {
+    String testPlaylistUrl =
+        "https://www.youtube.com/playlist?list=PLbpi6ZahtOH5AZzzL-piQQ2ryL6zitgqZ";
+    String playlistId = "PLbpi6ZahtOH5AZzzL-piQQ2ryL6zitgqZ";
+    Assert.assertEquals(playlistId, youtubeParser.getPlaylistIdFromUrl(testPlaylistUrl));
+
+    testPlaylistUrl = "https://www.youtube.com/playlist?list=PLbpi6ZahtOH784jF-_UIbH2cT9l1C_ecA";
+    playlistId = "PLbpi6ZahtOH784jF-_UIbH2cT9l1C_ecA";
+    Assert.assertEquals(playlistId, youtubeParser.getPlaylistIdFromUrl(testPlaylistUrl));
+  }
+
+  @Test
+  public void handleNonPlaylistLink() {
+    String testPlaylistUrl =
+        "https://www.youtube.com/watch?v=jxd42A3xFPs&list=PLbpi6ZahtOH784jF-_UIbH2cT9l1C_ecA&index=3&t=0s&app=desktop";
+    thrown.expect(IllegalArgumentException.class);
+    thrown.expectMessage(testPlaylistUrl + " is not a valid YouTube Playlist URL.");
+    youtubeParser.getPlaylistIdFromUrl(testPlaylistUrl);
+
+    testPlaylistUrl = "https://youtu.be/jxd42A3xFPs";
```

Comment:
I think it's okay to keep this second case here, since it's significantly different than the first case in this method